### PR TITLE
[SDK] Change: Migrate /linodes services to SDK

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Migrates all `/linodes` services and types.

Sorry about the size 🙏

### CODE REVIEW

File: packages/linode-js-sdk/src/linodes/types.ts [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+  label: string;
+  type: null | string;
+  status: LinodeStatus;
+  updated: string;
+  hypervisor: Hypervisor;
+  specs: LinodeSpecs;
+  watchdog_enabled: boolean;
+  tags: string[];
+  /** Added by UI */
+  recentEvent?: Event;
```
Comment (Jskobos) [inline packages/linode-js-sdk/src/linodes/types.ts]: This is a Cloud-specific functionality; should probably be split off and have an ExtendedLinode type that stays in Manager.

Comment (martinmckenna) [reply packages/linode-js-sdk/src/linodes/types.ts]: yep. We have to fix these for more than just Linodes. I have a separate ticket open to fix all types that are altered by the UI after things are migrated.

Comment (martinmckenna) [reply packages/linode-js-sdk/src/linodes/types.ts]: I imagine we'll just keep a `types` dir which will contain all the base interfaces, extended.

**Comment by martinmckenna** [comment]
@Jskobos addressed

**Pull request merged**